compiler/front_end/module_ir_test.py

```diff
@@ -4025,29 +3956,25 @@ def _check_source_location(source_location, path, min_start, max_end):
     result = []
     start = None
     end = None
-    if not source_location.HasField("start"):
+    if not source_location.start:
```

Comment:
`start` is always present now, but the __bool__ impl makes this check if it's a defaulted value, is that what we want? Should we maybe just add a `is_default()` method instead?

Reply:
This is mostly a matter of taste.  The `__bool__` method seems more Pythonic to me (truthy/falsy values rather than explicit tests), but your preference should probably override here.

Reply:
One practical effect of using `__bool__` is that you can do:

```Python
if x.source_location:
    # ...
```

instead of:

```Python
if x.source_location is not None and not x.source_location.is_default():
    # ...
```

Reply:
We cleared this up, this is fine.